src/claimer.ts

```diff
@@ -110,18 +110,28 @@ export class Claimer {
 
     private async gatherUnclaimedInfo(validatorAddress: string, validatorInfo: ValidatorInfo): Promise<number[]> {
 
+      const ledger = (await this.api.derive.staking.account(validatorAddress)).stakingLedger
+      if (!ledger) {
+          throw new Error(`Could not get ledger for ${validatorAddress}`);
+      }
+
       const lastReward = validatorInfo.lastReward
   
       const numOfPotentialUnclaimedPayouts = this.currentEraIndex - lastReward - 1;
       const unclaimedPayouts: number[] = []
+      const claimedRewards: number[] = ledger.claimedRewards.map(x => x.toNumber());
+
       for ( let i = 1; i <= numOfPotentialUnclaimedPayouts; i++) {
         const idx = lastReward + i;
+        if (claimedRewards.includes(idx)) {
```

Comment:
hey @mkaczanowski, thanks for the PR.
Could you elaborate a bit on why this would be necessary ? shouldn't the "validatorInfo.lastReward" be enough, why do you think it is not trustable ?

Reply:
hey, iirc (I made this PR a while ago) the `validatorInfo.lastReward` is not sufficient if you run with the `deepCheck`. It's fairly simple to repro this issue:
1. set deepcheck to true
2. start payouts and see that TXes will be created for the eras that have already been paid.

from what I see by quickly skimming through the code the deep check will make the payouts start prior to the `validatorInfo.lastReward`, thus create TXes that can only be rejected/fail.

This case never really worked even in the original repo:
github.com/w3f/polkadot-payouts

and I remember applying the same fix (among others I haven't published). But I would instead merge everything I have upstream :)